src/platform/options/default_configuration.cpp

```diff
@@ -303,19 +305,26 @@ void mo::DefaultConfiguration::parse_arguments(
         for (auto const& token : unparsed_arguments)
             tokens.push_back(token.c_str());
         if (!tokens.empty()) unparsed_arguments_handler(tokens.size(), tokens.data());
-
+        if (options.is_set("version"))
+        {
+            std::ostringstream mir_version;
+            mir_version << MIR_VERSION_MAJOR << "." << MIR_VERSION_MINOR << "." << MIR_VERSION_MICRO << std::endl;
+            std::cout << mir_version.str();
+            exit(0);
+        }
         if (options.is_set("help"))
         {
             std::ostringstream help_text;
             help_text << desc;
-            BOOST_THROW_EXCEPTION(mir::ExitWithOutput(help_text.str()));
+            std::cout << help_text.str();
+            exit(0);
```

Comment:
Obviously, most things ought to be cleaned up on exit, but I don't think there are enough guarantees about the state of the program when these options are processed.

It is cleaner to unwind the stack using the `mir::ExitWithOutput` exception that is intended for this purpose.

Reply:
Also tried to remove most duplicate logic that I could without making it entirely incomprehensible